packages/devtools_app/lib/src/config_specific/flutter/import_export/import_export.dart

```diff
@@ -49,6 +56,52 @@ class ImportController {
         return;
     }
   }
+
+  void _navigateToImportScreen(DevToolsScreenType screenType) {
+    final availableScreens = scaffoldState.widget.tabs;
+    int screenIndex =
+        availableScreens.indexWhere((screen) => screen.type == screenType);
+    if (screenIndex == -1) {
+      screenIndex = scaffoldState.widget.tabs.length;
+      scaffoldState.widget.tabs.add(screenType.create());
+    }
+    scaffoldState.tabController.animateTo(screenIndex);
+  }
+
+  void _importTimeline(Map<String, dynamic> json) async {
+    OfflineData offlineData;
+    final timelineMode =
+        json[TimelineData.timelineModeKey] == TimelineMode.full.toString()
+            ? TimelineMode.full
+            : TimelineMode.frameBased;
+    if (timelineMode == TimelineMode.frameBased) {
+      offlineData = OfflineFrameBasedTimelineData.parse(json);
+    } else {
+      offlineData = OfflineFullTimelineData.parse(json);
+    }
+
+    if (offlineData.isEmpty) {
+      _notifications.push('Imported file does not contain timeline data.');
```

Comment:
This is a very reasonable use of the Notifications class.  However, depending on all the flutter logic can make testing this class much more difficult (You'll have to use a widget test and pass in all the widget boilerplate just to get a controller to parse JSON).

When I originally sent out the Notifications class, I included a NotificationService interface that the class implemented.  This would allow us to have a service for tests that can surface notifications from controllers without needing the UI dependencies.

It wasn't needed at the time, but now that we have controllers surfacing notifications, this will make this class more testable.  I'll share that PR with you.

Reply:
added tests with NotificationService